src/contracts/interfaces/IStrategyFactory.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/proxy/beacon/IBeacon.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./IStrategy.sol";
+
+/**
+ * @title Interface for the `StrategyFactory` contract.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @dev This may not be compatible with non-standard ERC20 tokens. Caution is warranted.
+ */
+interface IStrategyFactory {
+
+    // @notice Upgradeable beacon which new Strategies deployed by this contract point to
+    function strategyBeacon() external view returns (IBeacon);
+
+    // @notice Mapping token => Strategy contract for the token
+    function tokenStrategy(IERC20 token) external view returns (IStrategy);
+
+    /**
+     * @notice Deploy a new strategyBeacon contract for the ERC20 token.
+     * @dev A strategy contract must not yet exist for the token.
+     * $dev Immense caution is warranted for non-standard ERC20 tokens, particularly "reentrant" tokens
+     * like those that conform to ERC777.
+     */
+    function deployNewStrategy(IERC20 token) external returns (IStrategy newStrategy);
+
+    /** 
+     * @notice Owner-only function to pass through a call to `StrategyManager.addStrategiesToDepositWhitelist`
+     * @dev Also adds the `strategiesToWhitelist` to the `tokenStrategy` mapping
+     */
+    function whitelistStrategies(
+        IStrategy[] calldata strategiesToWhitelist,
+        bool[] calldata thirdPartyTransfersForbiddenValues
+    ) external;
+
+    // @notice Owner-only function to add (existing) Strategy contracts to the `tokenStrategy` mapping
+    function editTokenStrategiesMapping(
+        IERC20[] calldata tokens,
+        IStrategy[] calldata strategies
+    ) external;
+
+    // @notice Owner-only function to modify the `strategyBeacon`
+    function setStrategyBeacon(IBeacon _strategyBeacon) external;
```

Comment:
open question: do we need this function (and associated event) at all?

Reply:
setStrategyBeacon seems fine, the editTokenStrategiesMapping() is gunna blow up any audit because we can't guarantee the state at any given point. Even in the best case this seems like a huge footgun.

Reply:
using `setStrategyBeacon` could/would result in a partition of the factory-deployed contracts into different sets that get upgraded via calling different beacons. I asked this because I'm just really not seeing a sensible use for it; it made more sense with the TUP pattern I had before.

I'm totally fine with removing `editTokenStrategiesMapping` -- I agree the contract would have easier-to-understand, better formally-defined behavior without it. It's essentially a governance-controlled "make an irregular state change" button, with limited practical use (but we can discuss more what I had in mind?), so I'm fine just getting rid of it

Reply:
I decided to remove this function in https://github.com/Layr-Labs/eigenlayer-contracts/pull/522/commits/b88cdb5617d8ba51b15b6472f655d8043d1929ee
I'm not seeing a practical use for it